tests/iscsi/iscsi_client.pm

```diff
@@ -8,58 +8,140 @@
+    # next and press connect button
+    send_key "alt-n";
+    assert_and_click 'iscsi-initiator-connect-button';
     assert_screen 'iscsi-initiator-connect-manual';
-    send_key "alt-n";                                     # go to connected targets tab
+    send_key 'alt-o';
+    assert_screen 'iscsi-initiator-discovery-enable-login-auth';
+    send_key 'alt-u';
+    type_string_slow_extended $test_data->{initiator_conf}->{user};
+    assert_screen 'iscsi-initiator-discovery-auth-initiators-username';
+    send_key 'alt-p';
+    my $init_pass = reverse $test_data->{common}->{password};
+    wait_screen_change { type_string_slow_extended $init_pass; };
+    send_key 'alt-r';
+    type_string_slow_extended $test_data->{target_conf}->{user};
+    assert_screen 'iscsi-initiator-discovery-auth-targets-username';
+    send_key 'alt-a';
+    wait_screen_change { type_string_slow_extended $test_data->{common}->{password}; };
+    send_key 'alt-n';
+}
+
+sub initiator_connected_targets_tab {
+    # go to discovered targets tab
+    send_key "alt-d";
     assert_screen 'iscsi-initiator-discovered-targets';
-    send_key "alt-n";                                     # next
+    # go to connected targets tab
+    send_key "alt-n";
     assert_screen 'iscsi-initiator-connected-targets';
-    send_key "alt-o";                                     # OK
+    # press OK twice
+    send_key "alt-o";
     wait_still_screen(2, 10);
+    send_key "alt-o";
+}
+
+
+sub run {
+    my $self = shift;
+    prepare_xterm_and_setup_static_network(ip => $test_data->{initiator_conf}->{ip}, message => 'Configure MM network - client');
+    mutex_wait('iscsi_target_ready', undef, 'Target configuration in progress!');
+    record_info 'Target Ready!', 'iSCSI target is configured, start initiator configuration';
+    my $module_name = $self->launch_yast2_module_x11('iscsi-client', target_match => 'iscsi-client');
+    initiator_service_tab;
+    initiator_discovered_targets_tab;
+    initiator_connected_targets_tab;
     wait_serial("$module_name-0", 180) || die "'yast2 iscsi-client ' didn't finish or exited with non-zero code";
-    assert_script_run 'lsscsi';
-    assert_script_run "echo -e \"n\\np\\n1\\n\\n\\nw\\n\" \| fdisk /dev/sda";    # create one partition
-    assert_script_run 'mkfs.ext4 /dev/sda1';                                     # format partition to ext4
-    assert_script_run 'mount /dev/sda1 /mnt';                                    # mount partition to /mnt
-    assert_script_run 'echo "iscsi is working" > /mnt/iscsi';                    # write text to file on iscsi disk
-    assert_script_run 'grep "iscsi is working" /mnt/iscsi';                      # grep expected text from file
+    # logging in to a target will create a local disc device
+    # it takes a moment, since udev actually handles it
+    sleep 5;
+    record_info 'Systemd', 'Verify status of iscsi services and sockets';
+    systemctl("is-active iscsid.service", expect_false => 0);
```

Comment:
`expect_false` is false by default. What is the motivation to specify it explicitly here?

Reply:
Frankly, I do not remember well. I know that I was not sure what kind of services should be running, so I guess it is a leftover from the times when I was just playing with the services. I am about to remove it.